scripts/pi-hole/php/func.php

```diff
@@ -180,17 +180,13 @@ function getCustomDNSEntries()
     if ($handle)
     {
         while (($line = fgets($handle)) !== false) {
-            $line = str_replace("\r","", $line);
-            $line = str_replace("\n","", $line);
-            $explodedLine = explode (" ", $line);
-
-            if (count($explodedLine) != 2)
-                continue;
-
-            $data = new \stdClass();
-            $data->ip = $explodedLine[0];
-            $data->domain = $explodedLine[1];
-            $entries[] = $data;
+            if (preg_match('/^([a-f0-9.:]+)\h+([^\#\s]+)/', $line, $match))
```

Comment:
I see you check for `#` also in the domain name, instead of only at the line beginning. Do you have any specific reason for doing so?

That said, I'm just curious and think this can be merged right away.

Reply:
It matches everything except `#` (start of comment) and `\s` (whitespace) to allow for inline comments. Looking at this, I don't understand why I even escaped the `#`.
Also just noticed that this could've been used in other places in that file as well (e.g. `getCustomCNAMEEntries()`). I think I'm just gonna close this as I really did a poor job. I was trying to solve the specific problem I had, but I don't even use `custom.list` anymore and this could've been done better.